### PR TITLE
[Fixes: #12448] Fix unread indicator in chat on message delete

### PR DESCRIPTION
**Pull request by cammellos**
When a message is deleted, we mark it as read.

### CODE REVIEW

File: src/status_im/chat/models.cljs [hunk 1]
```diff
@@ -431,3 +431,16 @@
...
+
+(fx/defn decrease-unviewed-count
+  {:events [:chat/decrease-unviewed-count]}
+  [{:keys [db]} chat-id unseen-messages]
+  {:db (update-in db [:chats chat-id :unviewed-messages-count]
+                  (fn [count]
+                    ;; There might be some other requests being fired,
+                    ;; so we need to make sure the count has not been set to
+                    ;; 0 in the meantime
+                    (let [new-count (- count unseen-messages)]
```
Comment (rasom) [inline src/status_im/chat/models.cljs]: could it be just `(max (- count unseen-messages) 0)`?

**Comment by churik** [comment]
### ISSUE 1: deleted messages are still shown on chat view until reopening view

**Steps:**
1) start 1-1 chat
2) send message  
3) long tap > delete it

**Expected result:** chat view is updated 

**Actual result:** still can see deleted message

**OS:** IOS, Android

https://user-images.githubusercontent.com/4557972/130468665-0d9a461e-a102-4aa8-a51f-66c80e598d66.mp4

**Review by rasom — APPROVED** [review]

File: src/status_im/chat/models.cljs (hunk 1 above)
Comment (cammellos) [reply src/status_im/chat/models.cljs]: yes, much better, thanks

**Comment by cammellos** [comment]
@churik should be fixed now

**Comment by churik** [comment]
@cammellos thank you for the fix, several small issues:
### ISSUE 2:Unread indicator in activity centre is not cleared in case of deleted message

**Steps:**
1) User A, B: join same public chat
2) User A: mention User B in public chat
3) User A: delete message
4) User B: check Acitivity centre

**Expected result:**
no unread indicator on AC, no message in AC

**Actual result:**
unread indicator is shown on AC, but it is empty when opened
**OS:** IOS, Android

### ISSUE 3: Unread indicator is not cleared for community mentions
**Steps:**
1) User A, B: join same community
2) User A: mention User B in community
3) User A: delete message

**Expected result:**
unread `1` is not shown on community
**Actual result:**
IMAGE 2021-08-24 15:44:00

**OS:** IOS, Android

Let me know if some of them should be fixed separately

**Pull request merged**